src/main/webapp/app/overview/courses.component.ts

```diff
@@ -88,11 +91,18 @@ export class CoursesComponent implements OnInit, OnChanges, OnDestroy {
                         });
                     }
                 });
-
-                this.nextRelevantExams = this.exams.filter(
-                    (exam) => this.serverDateService.now().isBefore(exam.endDate!) && this.serverDateService.now().isAfter(exam.visibleDate!),
-                );
-                this.nextRelevantExercise = this.findNextRelevantExercise();
+                this.nextRelevantExams = this.exams.filter((exam) => {
+                    if (exam.testExam!) {
```

Comment:
```suggestion
                    if (!!exam.testExam) {
```
If the value is undefined, this would lead to a type error, otherwise. `!!` evaluates to true, if the value is defined and true.

Reply:
`exam.testExam` is always defined, as it is enforced during creation / update of an exam, and it is always sent to the client when getting an exam from the client. 
Thanks for the good catch, we'll consider changing the entity model from a `?: boolean`  to a `: boolean`, as we also use a `boolean` on the server-side (instead of `Boolean`)

Reply:
Then it should not lead to an error I guess. It may be a good idea to change this in the future.

Reply:
> Thanks for the good catch, we'll consider changing the entity model from a ?: boolean to a : boolean, as we also use a boolean on the server-side (instead of Boolean)

You can do this in this PR i guess since i had also some remarks :+1: